web-local/src/lib/components/navigation/sources/SourceMenuItems.svelte

```diff
@@ -125,21 +125,66 @@
     }
   };
 
-  const bootstrapDashboard = async (id: string, tableName: string) => {
-    const previousActiveEntity = $rillAppStore?.activeEntity?.type;
-    const createdMetricsId = await autoCreateMetricsDefinitionForSource(
-      $persistentModelStore.entities,
-      $derivedTableStore.entities,
-      sourceID,
-      tableName
-    );
-
-    navigationEvent.fireEvent(
-      createdMetricsId,
-      BehaviourEventMedium.Menu,
-      MetricsEventSpace.LeftPanel,
-      EntityTypeToScreenMap[previousActiveEntity],
-      MetricsEventScreenName.Dashboard
+  const createDashboardFromSource = async (sourceName: string) => {
+    // create model from source
+    const newModelName = getName(`${sourceName}_model`, $modelNames.data);
+    $createFileMutation.mutate(
+      {
+        data: {
+          instanceId: $runtimeStore.instanceId,
+          path: `models/${newModelName}.sql`,
+          blob: `select * from ${sourceName}`,
+          create: true,
+          createOnly: true,
+          strict: true,
+        },
+      },
+      {
+        onSuccess: () => {
+          // create dashboard from model
+          const newDashboardName = getName(
+            `${newModelName}_dashboard`,
+            $dashboardNames.data
+          );
+          $createFileMutation.mutate(
+            {
+              data: {
+                instanceId: $runtimeStore.instanceId,
+                path: `dashboards/${newDashboardName}.yaml`,
+                blob: metricsTemplate, // TODO: compile a real yaml file
+                create: true,
+                createOnly: true,
+                strict: false,
+              },
+            },
```

Comment:
Can also create a reusable method for this, since it's being used at 3 places - 

```
export async function autoGenerateDashboardArtifact(
  instanceId: string,
  model: V1Model,
  putAndReconcileMutation: UseMutationResult<V1PutFileAndReconcileResponse>,
  dashboardNames: Array<string>
) {
  let metricsLabel = `${model?.name}_dashboard`;
  metricsLabel = getName(metricsLabel, dashboardNames);
  const generatedYAML = generateMeasuresAndDimension(model, {
    display_name: metricsLabel,
  });

  await putAndReconcileMutation.mutateAsync({
    data: {
      instanceId,
      path: getFileFromName(metricsLabel, EntityType.MetricsDefinition),
      blob: generatedYAML,
      create: true,
    },
  });

  return metricsLabel;
}
```